File: Algorithms/Java/P560-Subarray-Sum-Equals-K.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int res = 0;
        for (int i = 0; i < nums.length; ++i) {
            int curr = 0;
            for (int j = i; j < nums.length; ++j) {
                curr += nums[j];
                if (curr == k)
                    res++;
            }
        }
        
        return res;
    }
}

File: algorithm/py3/p560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        cnt = 0
        if len(nums) == 1 and nums[0] == k:
            return 1

        for left in range(len(nums)):
            sum = 0
            for right in range(left, len(nums)):
                sum += nums[right]
                if sum == k:
                    cnt += 1
        return cnt